### PR TITLE
Task/Loading State

### LINKED ISSUE #53: Add loading state to subscriptions list page

### PR DESCRIPTION
**Pull request by ziggysauce**
Resolves #53 

## Changes/Additions
* Adding `loading` to the main `subscriptionList` component
  * Added timeouts for perceived loading since instant request/responses made the UI jumpy and look funky
* Made a few more minor UI updates

## Note
* I don't particularly love all of my nested ternary components.... perhaps could do for a refactor later down the road

### CODE REVIEW

File: client/src/components/subscription/subscriptionsList.js [hunk 1]
```diff
@@ -72,29 +92,39 @@ class SubscriptionsList extends React.Component {
...
-              <UpdateSubscription updateSubscription={this.handleUpdate} currentSubscriptions={subscriptions} prevSubscription={subscription} />
+              <UpdateSubscription updateSubscription={this.handleUpdate} toggleLoadingState={this.toggleLoadingState} currentSubscriptions={subscriptions} prevSubscription={subscription} />
               <button onClick={() => this.setState({ editingSubscription: null })} className="btn btn-link my-2" type="button">Cancel</button>
             </div>
             }
         </div>
       );
     });
 
+    const addSubScriptionTemplates = addingSubscription
```
Comment (jduong09) [inline client/src/components/subscription/subscriptionsList.js]: change addSubScriptionTemplates to addSubscriptionTemplates

```suggestion
    const addSubscriptionTemplates = addingSubscription
```

**Review by jduong09 — CHANGES_REQUESTED** [review]
Looks great! 
1. When the page loads. On first creation of a subscription, getting a warning message in console. I attached the warning message that I received. 
2. Can we show the user all the subscriptions after user creates || updates their subscription. I believe it's possible, passing the boolean value into the create/updateSubscription forms.

Screen Shot 2022-01-26 at 3 45 25 PM

File: client/src/components/subscription/subscriptionsList.js [hunk 2]
```diff
@@ -97,17 +102,25 @@ class SubscriptionsList extends React.Component {
...
             }
         </div>
       );
     });
 
-    const addSubScriptionTemplates = addingSubscription
+    const addSubscriptionTemplates = addingSubscription
       ? <div className="card p-3 m-2 d-flex flex-column">
-          <CreateSubscription addSubscription={this.handleUpdate} toggleLoadingState={this.toggleLoadingState} currentSubscriptions={subscriptions} />
+          <CreateSubscription addSubscription={this.handleUpdate}
```
Comment (jduong09) [inline client/src/components/subscription/subscriptionsList.js]: Should we keep syntax the same as UpdateSubscription, so put addSubscription on newline?
```suggestion
          <CreateSubscription 
            addSubscription={this.handleUpdate}
```

**Review by jduong09 — CHANGES_REQUESTED** [review]
PR Looks great!

**Comment by jduong09** [comment]
I'm ready to merge it in after that syntax change.

File: client/src/components/subscription/subscriptionsList.js (hunk 2 above)
Comment (ziggysauce) [reply client/src/components/subscription/subscriptionsList.js]: oh yeahh that's fine. haha this is where having prettier for auto-formatting comes in handy

**Review by jduong09 — APPROVED** [review]
Looks good!

**Pull request merged**